src/content_delegate.js

```diff
@@ -585,31 +588,193 @@ ContentDelegate.prototype.attachRecapLinkToEligibleDocs = function() {
+        }));
+        recap_link.insertAfter(this.links[i]);
       }
-      let href = `https://www.courtlistener.com/${result.filepath_local}`;
-      let recap_link = $('<a/>', {
-        'class': 'recap-inline',
-        'title': 'Available for free from the RECAP Archive.',
-        'href': href
-      });
-      recap_link.click($.proxy(this.handleRecapLinkClick, this, window, href));
-      recap_link.append($('<img/>').attr({
-        src: chrome.extension.getURL('assets/images/icon-16.png')
-      }));
-      recap_link.insertAfter(this.links[i]);
+    }, this));
+};
+
+// TODO: implement appellate court check
+// TODO: Confirm that zip downloading is consistent across jurisdictions
+ContentDelegate.prototype.onDownloadAllSubmit = async function (event) {
+  // helper function - extract the zip by creating an html Element and then querying for the frame
+  const extractUrl = (html) => {
+    const page = document.createElement("html");
+    page.innerHTML = html;
+    const frames = page.querySelectorAll("iframe");
+    return frames[0].src;
+  };
+
+  // helper function - convert string to html document
+  const stringToDocBody = (str) => {
+    const parser = new DOMParser();
+    const newDoc = parser.parseFromString(str, 'text/html');
+    return newDoc.body;
+  };
+
+  // helper function - returns filename based on user preferences
+  const generateFileName = (options, pacerCaseId) => {
+    if (options.ia_style_filenames) {
+      return [
+        'gov',
+        'uscourts',
+        this.court,
+        (pacerCaseId || 'unknown-case-id')
+      ].join('.').concat('.zip');
+    } else if (options.lawyer_style_filenames) {
+      const firstTable = document.getElementsByTagName('table')[0];
+      const firstTableRows = firstTable.querySelectorAll('tr');
+      // 4th from bottom
+      const matchedRow = firstTableRows[firstTableRows.length - 4];
+      const cells = matchedRow.querySelectorAll('td');
+      const document_number = cells[0].innerText.match(/\d+(?=\-)/)[0];
+      const docket_number = cells[1].innerText;
+      return [
+        PACER.COURT_ABBREVS[this.court],
+        docket_number,
+        document_number,
+      ].join('_').concat('.zip');
+    }
+  };
+
+  // Make the Back button redisplay the previous page.
+  window.onpopstate = function (event) {
+    if (event.state.content) {
+      document.documentElement.innerHTML = event.state.content;
+    }
+  };
+  history.replaceState({content: document.documentElement.innerHTML}, '');
+
+  try {
+    // runtime start
+    $("body").css("cursor", "wait");
+    // fetch the html page which contains the <iframe> link to the zip document.
+
+    // in Firefox, use content.fetch for content-specific fetch requests
+    // https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts#XHR_and_Fetch
+    const browserSpecificFetch = (navigator.userAgent.indexOf('Chrome') < 0) ? content.fetch : window.fetch
```

Comment:
Should this be pulled out and generalized somewhere for future intrepid developers to not feel crazy?